Script/AtomicEditor/hostExtensions/languageExtensions/TypscriptLanguageExtension.ts

```diff
@@ -337,6 +341,29 @@ export default class TypescriptLanguageExtension implements Editor.HostExtension
         WebView.WebBrowserHost.setGlobalStringProperty("TypeScriptLanguageExtension", "tsConfig", JSON.stringify(tsConfig));
     }
 
+    generateExternalProject ()
+    {
+      var projectDir : string = ToolCore.toolSystem.project.projectPath;
+
+      // Create the typings folder in project root
+      var projectDirTypings : string = Atomic.addTrailingSlash(projectDir + "typings/main/ambient/atomicgameengine");
+      Atomic.getFileSystem().createDir(projectDirTypings);
+
+      // Copy the Atomic.d.ts definition file to the typings folder
+      var toolDataDir : string = ToolCore.toolEnvironment.toolDataDir;
+      var typescriptSupportDir : string = Atomic.addTrailingSlash(toolDataDir + "TypeScriptSupport");
+      Atomic.getFileSystem().copy(typescriptSupportDir + "Atomic.d.ts", projectDirTypings + "Atomic.d.ts");
+
+      // Generate a tsconfig.json file
+      defaultCompilerOptions.noLib = false;
+      var tsconfigFile = new Atomic.File(projectDir + "tsconfig.json", Atomic.FILE_WRITE);
+      let tsconfig = {
+        compilerOptions: defaultCompilerOptions
+      };
+      tsconfigFile.writeString(JSON.stringify(tsconfig));
```

Comment:
Sorry didn't catch this before, but the JSON for this file should probably be pretty printed so it's easy to edit.  That can be done with:
```JSON.stringify(tsconfig, null, 4)```  which tells the JSON process to format each entry correctly using 4 space tabs.